source/2_platform/rex_directx/include/rex_directx/dxgi/adapter.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "rex_directx/dxgi/comobject.h"
+#include "rex_directx/wrl/wrl_types.h"
+#include "rex_engine/types.h"
+#include "rex_renderer_core/gpu.h"
+
+#include <memory>
+#include <string>
+
+struct IDXGIAdapter;
+
+namespace rex
+{
+  namespace dxgi
+  {
+    class Adapter : public Gpu, public dxgi::ComObject<IDXGIAdapter> // NOLINT(fuchsia-multiple-inheritance)
+    {
```

Comment:
I realise that you use double inheritance here and want the functionality of the com object as well, I'm just wondering if there's a better, less complex way of designing it